routes/user.go

```diff
@@ -1215,6 +1215,8 @@ type GetSingleProfileRequest struct {
 	Username string `safeForLogging:"true"`
 	// When true, we don't log a 404 for missing profiles
 	NoErrorOnMissing bool `safeForLogging:"true"`
+	// When set, we filter extra data to return only keys that start with ExtraDataKeyPrefix.
+	ExtraDataKeyPrefix string `safeForLogging:"true"`
```

Comment:
what if we want to pull in data from multiple prefixes? are profile entries the only entity that uses extra data, and if not, should this be ProfileEntryExtraDataKeyPrefix?

Reply:
If multiple prefixes, best to use the shared common prefix, or just make two calls. Note this is mostly a perf improvement of reducing number of bytes over the wire when the extra data gets big in the future.

All transactions have extra data (but not all API shows it), but I think given that extra data is a top-level attribute to profile entry response, and that is also what we are filtering, that the ProfileEntry prefix is unneeded here.